marlowe-playground-client/src/Marlowe/Linter.purs

```diff
@@ -120,12 +122,15 @@ getWarningRange (SimplifiableValue range _ _) = range
 
 getWarningRange (SimplifiableObservation range _ _) = range
 
+getWarningRange (PayBeforeDeposit range _) = range
+
 newtype State
   = State
   { holes :: Holes
   , maxTimeout :: MaxTimeout
   , letBindings :: Set ValueId
   , warnings :: Set Warning
+  , deposits :: Set AccountId
```

Comment:
This needs to be in `LinEnv`, otherwise deposits in one branch will affect payments in another, I think. I am not sure why there is a copy of `letBindings` in here, it probably is not used because it is also defined in `LinEnv`, I didn't notice before :/